### ISSUE TITLE
[SUGGESTION] Bring back Analyze Solution

### ISSUE THREAD

**Issue by arkalyanms** [dotnet/vscode-csharp#6147]
From vscode-dotnettools created by [omlnaut](https://github.com/omlnaut): microsoft/vscode-dotnettools#358  

### Describe the feature you'd like

Currently, the problems tab only shows problems for files that were opened in the workspace. In the existing c# plugin, you could run Omnisharp: Analyze Current Project (or solution) to show all problems.

I need that feature back because it greatly simplifies fixing warnings in large projects

**Comment by mavasani** [comment]
From #5893

## Is your feature request related to a problem? Please describe.
Equivalent of `Run Code Analysis` command in VS to manually execute code analysis on demand for a project/solution. See https://learn.microsoft.com/en-us/visualstudio/code-quality/how-to-run-code-analysis-manually-for-managed-code for details

## Describe the solution you would like
- Add a new solution explorer command on project and solution nodes to trigger code analysis in background
- This may need some work at the Roslyn/LSP API layer to allow querying diagnostics for a specific project/solution

**Comment by reflectronic** [comment]
> Currently, the problems tab only shows problems for files that were opened in the workspace. 

To be clear @arkalyanms, this is an option that you can change: 
![image](https://github.com/dotnet/vscode-csharp/assets/27514983/ea0d8495-66d7-4b9d-8dfb-8a78ee0c789f)

**Comment by omlnaut** [comment]
@reflectronic not quite. This option runs the analyzer on the whole solution in the background permanently. This is using a full cpu core for me, all the time. I'd prefer a one-time-use command to run the analysis when I actually need it

**Comment by blushingpenguin** [comment]
that doesn't work, it never shows any analysis issues

**Comment by peterthorpe81** [comment]
> > Currently, the problems tab only shows problems for files that were opened in the workspace.
> 
> To be clear [@omlnaut](https://github.com/omlnaut), this is an option that you can change: ![image](https://github.com/dotnet/vscode-csharp/assets/27514983/ea0d8495-66d7-4b9d-8dfb-8a78ee0c789f)

This continuously runs the analyzer in the background which is incredibly resource intensive. 

In a blazor project its unusable, the error count constantly changes, I suspect due to source generated files.
